File: models/trigger.py
class Trigger:
    LESS = 'LESS'
    GREATER = 'GREATER'
    EQUAL = 'EQUAL'

    def __init__(self, threshold: float, mode: str = LESS):
        self.threshold = threshold
        self.mode = mode

    def check(self, value: float) -> bool:
        if self.mode == Trigger.LESS:
            return value < self.threshold
        elif self.mode == Trigger.GREATER:
            return value > self.threshold
        elif self.mode == Trigger.EQUAL:
            return abs(value - self.threshold) < 1e-4
        return False

    def to_dict(self):
        return {"threshold": self.threshold, "mode": self.mode}

    @classmethod
    def from_dict(cls, data):
        return cls(threshold=data["threshold"], mode=data["mode"])

    def __repr__(self):
        return f"Trigger(threshold={self.threshold}, mode={self.mode})"

File: java/evolution/src/main/java/com/life/evolution/model/Trigger.java
package com.life.evolution.model;

import java.util.HashMap;
import java.util.Map;

public class Trigger {

    public static final String LESS = "LESS";
    public static final String GREATER = "GREATER";

    private double threshold;
    private String mode;

    public Trigger(double threshold, String mode) {
        this.threshold = threshold;
        this.mode = mode;
    }

    public Trigger(double threshold) {
        this(threshold, LESS);
    }

    // ----------------------------------------------------
    // CHECK CONDITION
    // ----------------------------------------------------
    public boolean check(double value) {
        return switch (mode) {
            case LESS -> value < threshold;
            case GREATER -> value > threshold;
            default -> false;
        };
    }

    // ----------------------------------------------------
    // JSON SERIALIZATION
    // ----------------------------------------------------
    public Map<String, Object> toMap() {
        Map<String, Object> m = new HashMap<>();
        m.put("threshold", threshold);
        m.put("mode", mode);
        return m;
    }

    @SuppressWarnings("unchecked")
    public static Trigger fromMap(Map<String, Object> m) {
        return new Trigger(
                ((Number) m.get("threshold")).doubleValue(),
                (String) m.get("mode")
        );
    }

    @Override
    public String toString() {
        return "Trigger(threshold=" + threshold + ", mode=" + mode + ")";
    }

    // ----------------------------------------------------
    // GETTERS / SETTERS
    // ----------------------------------------------------
    public double getThreshold() { return threshold; }
    public void setThreshold(double threshold) { this.threshold = threshold; }

    public String getMode() { return mode; }
    public void setMode(String mode) { this.mode = mode; }
}
